app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -46,12 +46,12 @@ def handle_valid_otp
     redirect_to after_sign_in_path_for(current_user)
   end
 
-  def handle_invalid_otp
+  def handle_invalid_otp(type: 'otp')
```

Comment:
I'll add a comment. I didn't want to rename the method to keep the changes small, but I agree this should be renamed in a future PR. There is more refactoring to done around 2FA confirmation.